<launch>
  <arg name="joy_vel_in"  default="input_joy/cmd_vel"/>
  <arg name="joy_vel_out" default="joy_vel"/>

  <arg name="cmd_vel_out" default="cmd_mux/cmd_vel"/>

  <arg name="config_locks"  default="$(find cmd_mux)/config/cmd_mux_locks.yaml"/>
  <arg name="config_topics" default="$(find cmd_mux)/config/cmd_mux_topics.yaml"/>

  <arg name="config_joy" default="$(find cmd_mux)/config/joystick.yaml"/>

  <node pkg="cmd_mux" type="cmd_mux" name="cmd_mux" output="screen">
    <remap from="cmd_vel_out" to="$(arg cmd_vel_out)"/>

    <rosparam file="$(arg config_locks)"  command="load"/>
    <rosparam file="$(arg config_topics)" command="load"/>
  </node>

  <node pkg="cmd_mux" type="twist_marker" name="twist_marker">
    <remap from="twist"  to="$(arg cmd_vel_out)"/>
    <remap from="marker" to="twist_marker"/>
  </node>

  <node pkg="cmd_mux" type="joystick_relay.py" name="joystick_relay" output="screen">
    <remap from="joy_vel_in"  to="$(arg joy_vel_in)"/>
    <remap from="joy_vel_out" to="$(arg joy_vel_out)"/>

    <rosparam file="$(arg config_joy)" command="load"/>
  </node>
</launch>

